core/chains/evm/logpoller/log_poller.go

```diff
@@ -881,9 +889,35 @@ func (lp *logPoller) PollAndSaveLogs(ctx context.Context, currentBlockNumber int
 	}
 }
 
+// Returns information about latestBlock, latestFinalizedBlock and finalityDepth
+// If finality tag is not enabled, latestFinalizedBlock will be empty and finalityDepth will be fetched from config.
+// Otherwise, latestFinalizedBlock will be fetched from RPC and finalityDepth will be calculated as latestBlockNumber - latestFinalizedBlockNumber
```

Comment:
I think we can simplify this signature by just returning (latestHead *evmtypes.Head, latestFinalizedBlockNumber int64, error). We don't need anything other than the number for the finalized block. Should simplify some downstream logic, in pretty much all places we want to compute the latest finalized block number anyways using the finality depth

Reply:
I intended to encapsulate `finalityDepth` computation and put it in a single place, because for `finalityTagEnabled` we return it as `lastBlock - finalizedBlock`, and for other cases, we just return `finalityDepth` defined in the chain toml. 

At this moment, `lp.latestBlock` is used in 3 different places in LogPoller. If we have a signature as you suggested, every call to this method would require additional call, e.g.

```go
latestBlock, lastFinalizedBlockNo, err := lp.latestBlocks(lp.ctx)
finalityDepth := computeFinalityDepth(latestBlock, lastFinalizedBlock)
....
```
> Should simplify some downstream logic, in pretty much all places we want to compute the latest finalized block number anyways using the finality depth

Could you please elaborate? I understand that by having 2 values out of 3, we can compute the 3rd one, but I'm not sure how that would make downstream logic simpler. Do you mean that we don't need in a downstream logic `finalityDepth` and we should use `lastFinalizedBlockNumber` instead?

Reply:
Okay, bro, I was wrong. After spending more time with code and diving deeper I think you are right, `lastFinalityBlockNumber` is all we need